spec.bs

```diff
@@ -2091,6 +2162,402 @@ a {{ReportingBrowserSignals}} |browserSignals|, and a [=direct from seller signa
+<div algorithm="createAuctionNonce()">
+The <dfn for=Navigator method>createAuctionNonce()</dfn> method steps are:
+
+  1. Let |p| be [=a new promise=].
+  1. Run the following steps [=in parallel=]:
+    1. Let |nonce| be the [=string representation=] of a [=version 4 UUID=].
+
+    <div class=note>
+    Because we're going in parallel:
+      * There is no guarantee that the promise will be resolved before other tasks get queued on the
+        main thread;
+      * ...which gives browsers the freedom to generate this UUID in another process, and
+        asynchronously send it back to the main thread at an arbitrary future time.
+    </div>
+    1. [=Queue a task=] to [=resolve=] |p| with |nonce|.
+  1. Return |p|.
+</div>
+
+## Additional Bids ## {#additional-bids}
+
+*This first introductory paragraph is non-normative.*
+
+In addition to [=generate a bid|bids generated by interest groups=], sellers can enable buyers to
+introduce bids generated outside of the auction, which are called additional bids. Additional bids
+are commonly triggered using contextual signals. Buyers compute the additional bids, likely as part
+of a contextual auction. Buyers need to package up each additional bid using a new data structure
+that encapsulates all of the information needed for the additional bid to compete against other bids
+in a [[#running-ad-auctions|Protected Audience auction]].
+
+<div id="additional-bid-example" class=example>
+  <p> Each additional bid is expressed using the following JSON data structure:</p>
+  <pre class="highlight">
+  const additionalBid = {
+    "bid": {
+      "ad": 'ad-metadata',
+      "adCost": 2.99,
+      "bid": 1.99,
+      "bidCurrency": "USD",
+      "render": "https://www.example-dsp.com/ad/123.jpg",
+      "adComponents": [adComponent1, adComponent2],
+      "allowComponentAuction": true,
+      "modelingSignals": 123,
+    },
+    "interestGroup": {
+      "owner": "https://www.example-dsp.com"
+      "name": "campaign123",
+      "biddingLogicURL": "https://www.example-dsp.com/bid_logic.js"
+    },
+    "negativeInterestGroups": {
+      joiningOrigin: "https://www.example-advertiser.com",
+      interestGroupNames: [
+        "example_advertiser_negative_interest_group_a",
+        "example_advertiser_negative_interest_group_b",
+      ]
+    },
+    "auctionNonce": "12345678-90ab-cdef-fedcba09876543210",
+    "seller": "https://www.example-ssp.com",
+    "topLevelSeller": "https://www.another-ssp.com"
+  }
+  </pre>
+</div>
+
+<div algorithm>
+To <dfn>validate and convert additional bids</dfn> given an [=auction config=] |auctionConfig|, an
+[=auction config=]-or-null |topLevelAuctionConfig|, and a [=negative target info=]
+|negativeTargetInfo|:
+
+  1. [=Assert=] that these steps are running [=in parallel=].
+  1. Let |auctionNonce| be the [=string representation=] of |auctionConfig|'s
+    [=auction config/auction nonce=].
+  1. Let |capturedAdditionalBidsHeaders| be [=this=]'s [=relevant global object=]'s

```

Comment:
Can't call `[=this=]` from in parallel. Please make the caller pass in a global, navigable, or something else where we can identify the right place to access this.